modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/Slice2DIndex.java

```diff
@@ -159,14 +175,17 @@ public synchronized Slice2DIndex getSlice2DIndex( int imageIndex ) throws IOExce
             long endDataPosition = raf.readLong();
 
             raf.seek(dataPosition);
-            int tIndex = raf.readInt();
-            int zIndex = raf.readInt();
+            int dimensions = raf.readInt();
```

Comment:
You are right. It would be necessary to delete the index and recreate it. I do not know how that can be avoided though.